src/pages/images/actions/forms/ImportImageFromTarballForm.tsx

```diff
@@ -0,0 +1,157 @@
+import { ChangeEvent, FC, useState } from "react";
+import { useEventQueue } from "context/eventQueue";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { createImageAlias, createImageFromTarball } from "api/images";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { Link, useNavigate } from "react-router-dom";
+import { getImageToTarballBody, humanFileSize } from "util/helpers";
+import ProgressBar from "components/ProgressBar";
+import { UploadState } from "types/storage";
+import Loader from "components/Loader";
+
+interface Props {
+  close: () => void;
+  project: string;
+}
+
+const ImportImageFromTarballForm: FC<Props> = ({ close, project }) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const navigate = useNavigate();
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+
+  const notifySuccess = () => {
+    const imported = <Link to={`/ui/project/${project}/images`}>imported</Link>;
+    toastNotify.success(<>Image {imported} from file.</>);
+  };
+
+  const changeFile = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      void formik.setFieldValue("fileList", e.target.files);
+    }
+  };
+
+  const formik = useFormik<{
+    alias?: string;
+    isPublic: boolean;
+    fileList: FileList | null;
+  }>({
+    initialValues: {
+      alias: "",
+      isPublic: false,
+      fileList: null,
+    },
+    validationSchema: Yup.object().shape({
+      alias: Yup.string(),
+    }),
+    onSubmit: (values) => {
+      if (values.fileList) {
+        createImageFromTarball(
+          getImageToTarballBody(values.fileList),
+          values.alias || "",
+          setUploadState,
+        )
+          .then((operation) => {
+            toastNotify.info(<>Creation of image from file started.</>);
+
+            eventQueue.set(
+              operation.metadata.id,
+              (event) => {
+                const fingerprint = event.metadata.metadata?.fingerprint ?? "";
+                if (values.alias) {
+                  void createImageAlias(fingerprint, values.alias);
+                }
+                close();
```

Comment:
The close should also be called in the failure case. It might be simplest to move the close call to the 4th argument of eventqueue.set, which is a callback that is called in both cases, on success and failure.